src/server/address_space_internal.cpp

```diff
@@ -570,17 +570,27 @@ namespace OpcUa
 
     NodeId AddressSpaceInMemory::GetNewNodeId(const NodeId& id)
     {
+      uint32_t idx;
       if (id == ObjectId::Null || id.IsNull())
       {
-        return OpcUa::NumericNodeId(++MaxNodeIdNum, DefaultIdx);
+        idx = DefaultIdx;
+      } else {
+        if (id.HasNullIdentifier())
+        {
+          idx = id.GetNamespaceIndex();
+        } else {
+          return id;
+        }
       }
 
-      if (id.HasNullIdentifier())
+      for (;;)
```

Comment:
It's probably not, but the infinite loop looks scary.
Would there be a way to std::find the solution?